src/analysis/processing/qgsalgorithmextractlabels.cpp

```diff
@@ -0,0 +1,461 @@
+    {
+      processLabel( layerId, context, label, settings );
+    }
+
+    void processLabel( const QString &layerId, QgsRenderContext &context, pal::LabelPosition *label, const QgsPalLayerSettings &settings )
+    {
+      if ( mFeedback->isCanceled() )
+      {
+        context.setRenderingStopped( true );
+      }
+
+      const QgsFeatureId fid = label->getFeaturePart()->featureId();
+      if ( ( settings.placement == QgsPalLayerSettings::Curved ||
+             settings.placement == QgsPalLayerSettings::PerimeterCurved )
+           && !mCurvedWarningPushed.contains( layerId ) )
+      {
+        mCurvedWarningPushed << layerId;
+        mFeedback->pushWarning( QObject::tr( "Curved placement not supported, skipping labels from layer %1" ).arg( mMapLayerNames.value( layerId ) ) );
+        return;
+      }
+
+      QgsTextLabelFeature *labelFeature = dynamic_cast<QgsTextLabelFeature *>( label->getFeaturePart()->feature() );
+      QgsPalLayerSettings labelSettings( settings );
+      const QMap< QgsPalLayerSettings::Property, QVariant > &dataDefinedValues = labelFeature->dataDefinedValues();
+
+      if ( dataDefinedValues.contains( QgsPalLayerSettings::MultiLineWrapChar ) )
+      {
+        labelSettings.wrapChar = dataDefinedValues.value( QgsPalLayerSettings::MultiLineWrapChar ).toString();
+      }
+      if ( dataDefinedValues.contains( QgsPalLayerSettings::AutoWrapLength ) )
+      {
+        labelSettings.autoWrapLength = dataDefinedValues.value( QgsPalLayerSettings::AutoWrapLength ).toInt();
+      }
+      const QString labelText = QgsPalLabeling::splitToLines( labelFeature->text( -1 ),
+                                labelSettings.wrapChar,
+                                labelSettings.autoWrapLength,
+                                labelSettings.useMaxLineLengthForAutoWrap ).join( "\n" );
+
+      QString labelAlignment;
+      if ( dataDefinedValues.contains( QgsPalLayerSettings::MultiLineAlignment ) )
+      {
+        labelSettings.multilineAlign = static_cast< QgsPalLayerSettings::MultiLineAlign >( dataDefinedValues.value( QgsPalLayerSettings::MultiLineAlignment ).toInt() );
+      }
+      switch ( labelSettings.multilineAlign )
+      {
+        case QgsPalLayerSettings::QgsPalLayerSettings::MultiRight:
+          labelAlignment = QStringLiteral( "right" );
+          break;
+
+        case QgsPalLayerSettings::QgsPalLayerSettings::MultiCenter:
+          labelAlignment = QStringLiteral( "center" );
+          break;
+
+        case QgsPalLayerSettings::QgsPalLayerSettings::MultiLeft:
+          labelAlignment = QStringLiteral( "left" );
+          break;
+
+        case QgsPalLayerSettings::QgsPalLayerSettings::MultiJustify:
+          labelAlignment = QStringLiteral( "justify" );
+          break;
+
+        case QgsPalLayerSettings::MultiFollowPlacement:
+          switch ( label->getQuadrant() )
+          {
+            case pal::LabelPosition::QuadrantAboveLeft:
```

Comment:
Would it be possible to get the vertical alignment as well?

Reply:
The algorithm hard-codes the anchor / text block alignment to be bottom left. Is that something we'd want to handle differently?